cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/DefaultSecureStoreServiceTest.java

```diff
@@ -0,0 +1,182 @@
+    List<SecureKeyListEntry> secureKeyListEntries = secureStoreService.list(NamespaceId.DEFAULT);
+    Assert.assertEquals(secureKeyListEntries.size(), 1);
+    Assert.assertEquals(secureKeyListEntries.get(0).getName(), KEY1);
+    Assert.assertEquals(secureKeyListEntries.get(0).getDescription(), DESCRIPTION1);
+    revokeAndAssertSuccess(secureKeyId1, ALICE, ImmutableSet.of(Action.ALL));
+    secureKeyListEntries = secureStoreService.list(NamespaceId.DEFAULT);
+    Assert.assertEquals(0, secureKeyListEntries.size());
+
+    // Give BOB read access and verify that he can read the stored data
+    SecurityRequestContext.setUserId(BOB.getName());
+    grantAndAssertSuccess(secureKeyId1, BOB, ImmutableSet.of(Action.READ));
+    Assert.assertArrayEquals(secureStoreService.get(secureKeyId1).get(), VALUE1.getBytes());
+    secureKeyListEntries = secureStoreService.list(NamespaceId.DEFAULT);
+    Assert.assertEquals(1, secureKeyListEntries.size());
+
+    // BOB should not be able to delete the key
+    assertAuthorizationFailure(new SecureStoreOperationExecutor() {
+      @Override
+      public void execute() throws Exception {
+        secureStoreService.delete(secureKeyId1);
+      }
+    }, "Bob should not be able to delete a key since he does not have ADMIN privileges on the key");
+
+    // Grant Bob ADMIN access and he should be able to delete the key
+    grantAndAssertSuccess(secureKeyId1, BOB, ImmutableSet.of(Action.ADMIN));
+    secureStoreService.delete(secureKeyId1);
+    Assert.assertEquals(0, secureStoreService.list(NamespaceId.DEFAULT).size());
+  }
+
+  private void assertAuthorizationFailure(SecureStoreOperationExecutor operation, String failureMsg) throws Exception {
+    try {
+      operation.execute();
+      Assert.fail(failureMsg);
+    } catch (UnauthorizedException expected) {
+      // expected
+    }
+  }
+
+  private void grantAndAssertSuccess(EntityId entityId, Principal principal, Set<Action> actions) throws Exception {
+    Set<Privilege> existingPrivileges = authorizer.listPrivileges(principal);
+    authorizer.grant(entityId, principal, actions);
+    ImmutableSet.Builder<Privilege> expectedPrivilegesAfterGrant = ImmutableSet.builder();
+    for (Action action : actions) {
+      expectedPrivilegesAfterGrant.add(new Privilege(entityId, action));
+    }
+    Assert.assertEquals(Sets.union(existingPrivileges, expectedPrivilegesAfterGrant.build()),
+                        authorizer.listPrivileges(principal));
+  }
+
+
+  private void revokeAndAssertSuccess(EntityId entityId, Principal principal, Set<Action> actions) throws Exception {
+    Set<Privilege> existingPrivileges = authorizer.listPrivileges(principal);
+    authorizer.revoke(entityId, principal, actions);
+    for (Action action : actions) {
+      existingPrivileges.remove(new Privilege(entityId, action));
+    }
+    Assert.assertEquals(existingPrivileges, authorizer.listPrivileges(principal));
+  }
+
+  private interface SecureStoreOperationExecutor {
```

Comment:
you may not need this. I had added it in the dataset tests, because I had to check for more than one exception depending on the scenario